lightning-js/ui-components
# fix: normalize styles

## Description

This PR updates our components to utilize the latest style API, as well as adding a base level prop interface for all of our components to implement

known issues: the `style` prop on Views and Texts is currently throwing a type error for props.style. the type needs to be updated to allow nested arrays

**update**: this also includes some type updates to the `chainFunctions` utility that should resolve the errors that we were attempting to resolve in PR https://github.com/lightning-js/ui-components/pull/103

## Changes
- replaces all instances of `...[props.style].flat()` with `props.style`, thanks to solid's latest update to accept nested style arrays
- updates component prop types to leverage the solid node types where applicable
- adds `UIComponentProps` interface to ensure all of our components implement the same standard props(tone, styles, a patch for color)
- components that accept children must now explicitly define the children type
- improves the type safety and overall resilience of the `withScrolling` utility

## Testing

- ensure all components still render as expected
- build the project and ensure it loads correctly in client applications
